### PR TITLE
sink(ticdc): rename eventsink and remove useless type declaration

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/8132

### What is changed and how it works?

- Rename eventsink to dmlsink
- Remove useless type declaration

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

**Review by hi-rustin — COMMENTED** [review]
Self check

File: cdc/owner/ddl_sink.go [hunk 1]
```diff
@@ -27,8 +27,8 @@ import (
 	"github.com/pingcap/tidb/parser/format"
 	"github.com/pingcap/tiflow/cdc/contextutil"
 	"github.com/pingcap/tiflow/cdc/model"
-	sinkv2 "github.com/pingcap/tiflow/cdc/sinkv2/ddlsink"
-	"github.com/pingcap/tiflow/cdc/sinkv2/ddlsink/factory"
+	sinkv2 "github.com/pingcap/tiflow/cdc/sink/ddlsink"
```
Comment (amyangfei) [inline cdc/owner/ddl_sink.go:30]: no need to alias to `sinkv2`

**Review by amyangfei — APPROVED** [review]
rest lgtm

**Review by hicqu — APPROVED** [review]

File: cdc/owner/ddl_sink.go (hunk 1 above)
Comment (hi-rustin) [reply cdc/owner/ddl_sink.go]: Good catch! all updated.

**Comment by hi-rustin** [comment]
Thanks for your review!  💚 💙 💜 💛 ❤️

**Comment by hi-rustin** [comment]
/run-kafka-integration-test 

network issue

**Pull request merged**